javascript/README.md

```diff
@@ -283,6 +300,285 @@ const { a, ...noA } = copy; // noA => { b: 2, c: 3 }
+  animalsCopy[i] = animals[i];
+}
+
+// Works but is not preferred
+const animals = ['ant', 'bison', 'camel', 'duck', 'elephant'];
+const animalsCopy = animals.slice();
+```
+
+🎉 Yep! 🎉
+
+```js
+const animals = ['ant', 'bison', 'camel', 'duck', 'elephant'];
+const animalsCopy = [...animals];
+```
+
+#### Resources:
+
+- JSPerf: [Shallow Copy Arrays]
+
+### 3.4 Arrays From Iterables
+
+To convert an iterable object (e.g. [NodeList]) to an array, use [array spread syntax][Array Literal Spread Syntax] `...` instead of [Array.from].
+
+> Why? Better performance.
+
+#### Examples
+
+🚫 Nope. 🚫
+
+```js
+const paragraphs = document.querySelectorAll('p');
+const nodes = Array.from(paragraphs);
+```
+
+🎉 Yep! 🎉
+
+```js
+const paragraphs = document.querySelectorAll('p');
+const nodes = [...paragraphs];
+```
+
+#### Resources
+
+- ESLint: [prefer-spread]
+- JSPerf: [Arrays From Iterables]
+
+### 3.5 Arrays from Array-Like Objects
+
+Use [`Array.from`][Array.from] for converting an array-like object to an array.
+
+> Why? Not only is it easier to read/type but it also performs better.
+
+#### Examples
+
+🚫 Nope. 🚫
+
+```js
+const arrLike = { 0: 'foo', 1: 'bar', 2: 'baz', length: 3 };
+const arr = Array.prototype.slice.call(arrLike);
+```
+
+🎉 Yep! 🎉
+
+```js
+const arrLike = { 0: 'foo', 1: 'bar', 2: 'baz', length: 3 };
+const arr = Array.from(arrLike);
+```
+
+#### Resources
+
+- JSPerf: [Arrays from Array-Like Objects]
+
+### 3.6 Mapping Over Iterables
+
+Use [array spread syntax][Array Literal Spread Syntax] `...` instead of [`Array.from`][Array.from] for mapping over iterables.
+
+> Why? Overall better performance.
+
+#### Examples
+
+🚫 Nope. 🚫
+
+```js
+const iterable = 'Hello there!';
+const upperCase = letter => letter.toUpperCase();
+const upperCaseLetters = Array.from(iterable, upperCase);
+```
+
+🎉 Yep! 🎉
+
+```js
+const iterable = 'Hello there!';
+const upperCase = letter => letter.toUpperCase();
+const upperCaseLetters = [...iterable].map(upperCase);
+```
+
+#### Resources
+
+- JSPerf: [Mapping Over Iterables]
+
+### 3.7 Array Callback Return
+
+Use `return` statements in array method callbacks. It’s okay to omit the `return` if the function body consists of a single statement returning an expression without side effects.
+<!-- @TODO: Link later: following [8.2](#arrows--implicit-return).  -->
+
+#### Examples
+
+🚫 Nope. 🚫
+
+```js
+inbox.filter(msg => {
+  const { subject, author } = msg;
+  if (subject === 'Mockingbird') {
+    return author === 'Harper Lee';
+  } else {
+    return false;
+  }
+});
+```
+
+🎉 Yep! 🎉
+
+```js
+inbox.filter(msg => {
+  const { subject, author } = msg;
+  if (subject === 'Mockingbird') {
+    return author === 'Harper Lee';
+  }
+
+  return false;
+});
+```
+
+🎉 Also good! 🎉
+
+```js
+[1, 2, 3].map((x) => {
+  const y = x + 1;
+  return x * y;
+});
+
+// The return can be omitted here.
+[1, 2, 3].map(x => x + 1);
+```
+
+#### Resources
+
+- ESLint: [array-callback-return]
+
+### 3.8 Array Line Breaks
```

Comment:
@calebeby @Josh68 Please also pay special attention to this section. I know we have some rules in place but I don't know if what I show here is accurate. Thanks!

Reply:
It appears that the rule is that line breaks, indents, etc are driven by the individual contents of the array, not the array itself, which defaults to no formatting (i.e., inline). https://eslint.org/docs/rules/array-element-newline and related rulesets apply, but I'm not sure what the best settings are.

Reply:
I'm tempted to remove the **3.8 Array Line Breaks** section. I'm not sure how helpful it would be to have "Code should conform to [Prettier](https://prettier.io/) formatting". 

Do you feel strongly otherwise, @calebeby? 🤔I'm open to be convinced if you feel there's a benefit. :)

Reply:
I think you should mention prettier _somewhere_ in the js guide, but I think it is fine if you remove that section and put the prettier note at the top or something.